src/Stratis.Bitcoin/P2P/Peer/NetworkPeerConnection.cs

```diff
@@ -154,8 +153,8 @@ private async Task ReceiveMessagesAsync()
             }
             catch (Exception ex) when (ex is IOException || ex is OperationCanceledException || ex is ObjectDisposedException)
             {
-                this.logger.LogDebug("Receiving cancelled. Exception: {0}", ex.ToString());
-                this.peer.Disconnect("Receiving cancelled.");
+                this.logger.LogDebug("Connection closed whilst receiving messages. May be a peer in discovery. Exception: {0}", ex.ToString());
```

Comment:
I'm wondering whether we should just rather state that the node stopped receiving messages from another peer and perhaps add a bit more detail?

this.logger.LogDebug("The node stopped receiving messages from {0} peer {1}; exception: {2}", this.peer.Inbound ? "inbound" : "outbound", this.peer.PeerEndPoint, ex.ToString());